### PR TITLE
Andrii/fix/chat auth

### PR DESCRIPTION
**Pull request by AndreyBerezyanskiy**

### CODE REVIEW

File: app/views/chat_auth.py [hunk 1]
```diff
@@ -154,277 +157,366 @@ def buy():
...
-        phone=app.config["CHAT_DEFAULT_PHONE"],
-        card=app.config["CHAT_DEFAULT_CARD"],
-        password="",
-    ).save(False)
-    db.session.flush()
-    room.seller_id = user.id
-    db.session.commit()
-    log(log.INFO, f"User {user_name} created")
+    assert user
+    assert room
```
Comment (Taraslysak) [inline app/views/chat_auth.py:234]: I don't like these assertions

File: app/views/chat_auth.py [hunk 2]
```diff
@@ -154,277 +157,366 @@ def buy():
...
+        return render_template(
+            "chat/registration/05_name.html",
+            error_message="Please, add your name",
+            room=room,
+            now=response.now_str,
+            user_unique_id=response.params.user_unique_id,
+        )
+
+    assert user
+    assert room
```
Comment (Taraslysak) [inline app/views/chat_auth.py:418]: Same as above

File: src/chat.ts [hunk 3]
```diff
@@ -168,37 +168,38 @@ if (categoryDropdowns) {
...
-      dropdownMenu.classList.toggle('dropdown-list-active');
-      dropdownArrow.classList.toggle('rotate-180');
-      categoryButtons.forEach(button => {
-        button.addEventListener('click', () => {
-          const currentCategory = button.getAttribute('data-category');
-          eventCategoryInput.value = currentCategory;
-          dropdownTitle.innerHTML = currentCategory;
-          dropdownMenu.classList.remove('dropdown-list-active');
-          dropdownArrow.classList.remove('rotate-180');
+    if (dropdownButton) {
```
Comment (Taraslysak) [inline src/chat.ts:171]: `if (!dropdownButton){return}` ?

File: src/utils.ts
Comment (Taraslysak) [inline src/utils.ts:1]: do we need empty file?

**Review by Taraslysak — COMMENTED** [review]
Please consider a minor refactoring:
1. Consider removing numbers from filenames
2. Consider removing assertions from non-test code base. They are redundant, since absent user or room should result in error response state anyway state. If you expect the opposite behavior, it feels like code is not test-covered enough

File: app/commands/__init__.py [hunk 4]
```diff
@@ -133,7 +141,7 @@ def delete_user(email: str):
                 db.session.delete(payment)
         db.session.delete(user)
         db.session.commit()
-        print(f"User {user.username} deleted")
+        print(f"User {user.email} deleted")
```
[the remark this answers is not in the document]
Comment (AndreyBerezyanskiy) [reply app/commands/__init__.py:144]: No, this is log  if the user has been deleted. If the user is not found, another log is displayed

File: app/views/chat_auth.py [hunk 5]
```diff
@@ -154,277 +157,366 @@ def buy():
...
+            "chat/registration/01_email.html",
+            error_message=response.message,
             room=room,
             now=now_str,
+            email_input=response.email,
         )
 
-    user_query = m.User.select().where(m.User.username == user_name)
-    user: m.User = db.session.scalar(user_query)
+    assert user
```
[the remark this answers is not in the document]
Comment (AndreyBerezyanskiy) [reply app/views/chat_auth.py:191]: There is a check if a user is not found in that function. And the result has an error key. The return from the function will be earlier, if there is no user. But Mypy forces me to add this check.

File: src/utils.ts
Comment (AndreyBerezyanskiy) [reply src/utils.ts:1]: removed

File: src/chat.ts (hunk 3 above)
Comment (AndreyBerezyanskiy) [reply src/chat.ts:171]: agree, it's better.

**Pull request merged**